xgi/generators/classic.py

```diff
@@ -196,17 +207,26 @@ def flag_complex(g, max_order=2, seed=None):
     """
     # This import needs to happen when this function is called, not when it is
     # defined.  Otherwise, a circular import error would happen.
-    from ..classes import SimplicialComplex
 
-    nodes = g.nodes()
-    edges = list(g.edges())
+    nodes = G.nodes()
+    edges = G.edges()
 
-    # compute all triangles to fill
-    max_cliques = list(nx.find_cliques(g))
+    # compute all cliques to fill
+    max_cliques = list(nx.find_cliques(G))
 
     S = SimplicialComplex()
     S.add_nodes_from(nodes)
-    S.add_simplices_from(max_cliques, max_order=max_order)
+    S.add_simplices_from(edges)
+    if not ps: # promote all cliques
+        S.add_simplices_from(max_cliques, max_order=max_order)
+    else: # promote cliques with a given probability
+        cliques_d = defaultdict(list)
+        for x in max_cliques:
+            cliques_d[len(x)].append(x)
+        for i, p in enumerate(ps[:max_order - 1]):
+            d = i + 2 # simplex order
+            cliques_d_to_add = [el for el in cliques_d[d+1] if seed.random() <= p]
+            S.add_simplices_from(cliques_d_to_add, max_order=max_order)
 
     return S
```

Comment:
This is just a tiny nitpick, feel free to ignore (I'm already accepting). I prefer this because in this way, the code that actually does most of the work ends up on a lower level of indentation so it's easier to read:

```suggestion
    if not ps: # promote all cliques
        S.add_simplices_from(max_cliques, max_order=max_order)
        return S
        
    # promote cliques with a given probability
    cliques_d = defaultdict(list)
    for x in max_cliques:
        cliques_d[len(x)].append(x)
            
    for i, p in enumerate(ps[:max_order - 1]):
        d = i + 2 # simplex order
        cliques_d_to_add = [el for el in cliques_d[d+1] if seed.random() <= p]
        S.add_simplices_from(cliques_d_to_add, max_order=max_order)

    return S
```